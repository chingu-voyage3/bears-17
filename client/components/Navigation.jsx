import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const LINKS = [
  { label: 'Home', href: '/' },
  { label: 'About', href: '/about' },
  { label: 'Questions', href: '/questions' },
  { label: 'Ask Question', href: '/create' },
  { label: 'Login', href: '/login' },
  { label: 'Register', href: '/register' },
  { label: 'Profile', href: '/profile' },
  { label: 'Logout', href: '/logout' },
];

const loggedInNavs = ['Ask Question', 'Profile', 'Logout'];
const loggedOutNavs = ['Login', 'Register'];

const linkFilter = (array1, array2) => (
  array1.filter(e => array2.indexOf(e.label) === -1)
);

const Navigation = (props) => {
  const NavLinks = props.loggedIn
    ? linkFilter(LINKS, loggedOutNavs)
    : linkFilter(LINKS, loggedInNavs);

  return (
    <nav aria-label="Site links">
      <ul className="main-nav">
        {NavLinks.map(link => (
          <li key={link.label} className="main-nav__item">
            <Link to={link.href} className="main-nav__link">{link.label}</Link>
          </li>
      ))}
      </ul>
    </nav>
  );
};

export default Navigation;

Navigation.defaultProps = {
  loggedIn: '',
};

Navigation.propTypes = {
  loggedIn: PropTypes.string,
};
