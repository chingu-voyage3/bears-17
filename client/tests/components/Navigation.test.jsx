import React from 'react';
import renderer from 'react-test-renderer';

import Navigation from 'Components/Navigation';

describe('Navigation', () => {
  it('should match the snapshot', () => {
    const nav = renderer.create(<Navigation />);

    expect(nav.toJSON()).toMatchSnapshot();
  });
});
