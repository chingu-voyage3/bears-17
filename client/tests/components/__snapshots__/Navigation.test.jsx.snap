// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match the snapshot 1`] = `
<nav
  aria-label="Site links"
>
  <ul
    className="main-nav"
  >
    <li
      className="main-nav__item"
    >
      <a
        className="main-nav__link"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        className="main-nav__link"
        href="/about"
      >
        About
      </a>
    </li>
    <li
      className="main-nav__item"
    >
      <a
        className="main-nav__link"
        href="/login"
      >
        Login
      </a>
    </li>
  </ul>
</nav>
`;
